pkg/images/enricher/enricher_impl.go

```diff
@@ -299,6 +309,10 @@ func (e *enricherImpl) EnrichImage(ctx context.Context, enrichContext Enrichment
 	e.cvesSuppressor.EnrichImageWithSuppressedCVEs(image)
 	e.cvesSuppressorV2.EnrichImageWithSuppressedCVEs(image)
 
+	if scanResult == ScanNotDone && shouldDelegate {
+		errorList.AddError(errCentralScan)
+	}
```

Comment:
Building on the previous comment, perhaps this can be simplified:

```suggestion
	if !errorList.Empty() {
		errorList.AddError(delegateErr)
	}
```

Reply:
I understand the previous comment:

> Thinking we may be able to reduce the 'lines' here (and tests for ... 

But not sure about this suggestion because it's not checking if delegated scanning is enabled here. Do you mind to explain a bit more?